src/modules/utils/readpin/ReadPin.cpp

```diff
@@ -0,0 +1,59 @@
+#include "libs/Kernel.h"
+#include "ReadPin.h"
+#include "libs/nuts_bolts.h"
+#include "libs/utils.h"
+#include "Config.h"
+#include "SlowTicker.h"
+#include "libs/SerialMessage.h"
+#include "libs/StreamOutput.h"
+#include "checksumm.h"
+#include "ConfigValue.h"
+#include "StreamOutputPool.h"
+
+using namespace std;
+
+#define gcode_checksum               CHECKSUM("gcode")
+#define pin_checksum                 CHECKSUM("pin")
+
+ReadPin::ReadPin() {
+}
+
+ReadPin::ReadPin(uint16_t name) {
+    name_checksum = name;
+}
+
+void ReadPin::on_module_loaded() {
+    THEKERNEL->slow_ticker->attach(5, this, &ReadPin::pin_tick);
+    register_for_event(ON_GCODE_RECEIVED);
+
+    // Settings
+    on_config_reload(this);
+}
+
+// Get config
+void ReadPin::on_config_reload(void*) {
+    my_pin.from_string(THEKERNEL->config->value(readpin_checksum, name_checksum, pin_checksum )->by_default("nc")->as_string())->as_input();
+    if (my_pin.connected())
+        my_pin.pull_none();
```

Comment:
@wolfmanjm you mean the pull_none() call? I kinda forgot about that, when I first tried the pin apparently had a pull-up.